openapi/src/main/java/io/helidon/openapi/Serializer.java

```diff
@@ -0,0 +1,349 @@
+            return super.representScalar(tag, value, isExemptedFromQuotes(tag) ? DumperOptions.ScalarStyle.PLAIN : style);
+        }
+
+        @Override
+        protected Node representSequence(Tag tag, Iterable<?> sequence, DumperOptions.FlowStyle flowStyle) {
+            Node result = super.representSequence(tag, sequence, flowStyle);
+            representedObjects.clear();
+            return result;
+        }
+
+        private boolean isExemptedFromQuotes(Tag tag) {
+            return tag.equals(Tag.BINARY) || tag.equals(Tag.BOOL) || tag.equals(Tag.FLOAT)
+                    || tag.equals(Tag.INT);
+        }
+
+        @Override
+        protected NodeTuple representJavaBeanProperty(Object javaBean, Property property, Object propertyValue,
+                Tag customTag) {
+            if (propertyValue == null) {
+                return null;
+            }
+
+            Property p = property;
+            Object v = adjustPropertyValue(propertyValue);
+            Class<?> type = implsToTypes.get(javaBean.getClass()).getType();
+            if (type.isEnum()) {
+                p = new DelegatingProperty(property, property.toString());
+            }
+            if (propertyValue instanceof Enum) {
+                Enum e = (Enum) propertyValue;
+                v = e.toString();
+            }
+            NodeTuple result = okToProcess(javaBean, property)
+                    ? super.representJavaBeanProperty(javaBean, p, v, customTag) : null;
+            return result;
+        }
+
+        private Object adjustPropertyValue(Object propertyValue) {
+            /* Some MP OpenAPI TCK tests expect an integer-style format, even for BigDecimal types, if the
+             * value is an integer. Because the formatting is done in SnakeYAML code based on the type of the value,
+             * we need to replace a, for example BigDecimal that happen to be an integer value, with an Integer.
+             * See https://github.com/eclipse/microprofile-open-api/issues/412
+             */
+            if (Number.class.isInstance(propertyValue) && !Boolean.getBoolean("io.helidon.openapi.skipTCKWorkaround")) {
+                Number n = (Number) propertyValue;
+                float diff = n.floatValue() - n.intValue();
+                if (diff == 0) {
+                    propertyValue = Integer.valueOf(n.intValue());
+                } else if (Math.abs(diff) < 0.1) {
+                    LOGGER.warning(String.format("Integer approximation of %f did not match but the difference was only %e",
+                            n, diff));
+                }
+            }
+            return propertyValue;
+        }
+
+        @Override
+        protected MappingNode representJavaBean(Set<Property> properties, Object javaBean) {
+            MappingNode result = super.representJavaBean(properties, javaBean);
+            processExtensions(result, javaBean);
+            representedObjects.clear();
```

Comment:
Maybe put in a `finally` block?

Reply:
Clearing `representedObjects` is an awkward but effective way to prevent SnakeYAML from using anchors and links in the output document (this is the least bad way, currently, BTW) because Jackson - which is used in the MP OpenAPI TCK - does not process those correctly. If some method invoked here throws an exception then all bets are off and we don't really care about the contents of `representedObjects` any more. I've added a comment explaining this.